### ISSUE TITLE
[Backport v2.5] s3 etcd backups taking longer than 5 minutes cause backup loop

### ISSUE THREAD

**Issue by rancherbot** [rancher/rancher#36038]
This is a backport issue for https://github.com/rancher/rancher/issues/34890, automatically created via rancherbot by @snasovich

Original issue description:

Backups that take longer than 5 minutes will cause Rancher to start a new backup and delete the one that is currently running. Since they all take longer than 5 minutes it will continue this cycle indefinitely.

**Rancher Server Setup**
- Rancher version: 2.6.3

**Repro steps:**
* Have RKE1 provisioned downstream cluster
* Fill etcd with lots of data (e.g. 1 GB+) - enough so it takes more than 5 minutes to complete on all of the etcd nodes.
* Configure recurring snapshots for this downstream cluster, including setting `rancher_kubernetes_engine_config.services.etcd.backup_config.timeout` to high enough value for backups to actually complete on etcd side (e.g. 900 for 15 minutes).
* Wait for a backup to start (depending on interval configured) and wait 5 minutes.

**Actual behavior:**
Since the backup doesn't complete before the sync interval, a new backup will be started. This will repeat indefinitely until a backup is completed in under 5 minutes.

**Expected behavior:**
Backups to complete.

Original description (no longer relevant as fix approach changed per https://github.com/rancher/rancher/issues/34890#issuecomment-943872638):
> The `clusterBackupCheckInterval` is hardcoded to 5 minutes and should be user-changeable via the following
> - Helm (likely)
> - ENV VAR (could be good idea)
> - ConfigMap (eh)
> - etc.
> 
> Moreover, we should considered adding a "maximum retry count" so that the check interval does not infinitely continue.

JIRA ID: SURE-3305

**Comment by slickwarren** [comment]
tested on rancher v2.5-head (bd840b434201ed280db8a55c46751815a6d1904f):
rancher setup:
* downstream cluster (single node, latest k8s) with backups configured to minio through proxy
  * I have the timeout in the cluster.yaml set to 1200 seconds
minio setup:
* I have an nginx proxy (limited at 10kBps upload/download) to a minio setup. So for a 3mb file that should take ~5 minutes
  * I can confirm this by downloading an existing backup through the proxy to see the speed

Issue:
* when I use this with etcd snapshots, it seems to still have a similar issue on 2.5-head and 2.5.12-rc2 as it does on 2.6.3. 
* If I set the speed to something not close to the 5 minute (existing, hardcoded) timeout, it is successful. i.e. 50k works fine.

2.5.11 vs 2.5-head:
* I now see Waiting on etcdbackup-controller message from rancher on the etcd backup when the limit is set to last longer that 5 mins.  I have only seen this on the fixed versions, **but the backup still never finishes.**

**Comment by slickwarren** [comment]
after some more testing, I was still unable to reproduce the original issue.  We currently don't have an exact way to saturate etcd to the level that the customer reported, so we tried the following workaround with no luck reproducing:

test 1 -- throttle upload to s3 by adding a proxy in front of the s3 server -- tested on on 2.5-head (923f59cb62731d275d050802ab403153eb37d8ac) and 2.6.3
* throttle connection to your s3 store to 20kbps
* set timeout to > 5 minutes (I used 3600 seconds)
* ensure that fresh install backup completes successfully
* saturate etcd with some secrets (~500)
* take another snapshot 
* RESULT: snapshot uploads to s3 store, but etcd snapshot remains in `activating` state through rancher 
  * persists through refreshes
  * persists after new timeout limit is reached
  * only way to get it out of this state that I found was to increase the connection throttle to a greater value
  * the following error is observed in the rancher logs: 
```
Waiting for [etcd-snapshot-once] container to exit on host [<rancher-hostname>]
Container [etcd-snapshot-once] is still running on host [<rancher-hostname>]: stderr: [time="<current_time>" level=info msg="failed to upload etcd snapshot file: 504 Gateway Time-out, retried 1 times"
```
which repeats indefinitely, from 0 to 1 to 2 `retried <> times`

Perhaps this is closer to what the customer encountered, as even after 5 minutes I am able to see the backup file is successfully uploaded in the s3 store on 2.6.3.

**Comment by sowmyav27** [comment]
PR: https://github.com/rancher/rancher/pull/36202

**Comment by slickwarren** [comment]
tested on 2.5-head (f63bb65c09c5c9232f6e8ee07259fb9054a3f334) with a throttled proxy to minio:
* deploy downstream cluster configured to backup to proxied minio (set backup timeout to 1 hour)
  * initial backup should complete in less than 5 mintues -- pass
  * saturate etcd, making backup take longer than 5 minutes, and perform manual backup -- pass
  * wait for automatic backup to begin and complete (automatic backup should take longer than 5 minutes) -- pass
  * restore from initial backup -- pass
  * restore from saturated backup -- pass

also found this issue when testing https://github.com/rancher/rancher/issues/36209
